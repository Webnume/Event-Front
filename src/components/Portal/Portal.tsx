import { useEffect } from "react";
import { createPortal } from "react-dom";

const Portal = ({ children }: any) => {
  const mount = document.getElementById("portal");

  const el = document.createElement("div");

  useEffect((): any => {
    if (!mount) return;
    mount.appendChild(el);
    return () => mount.removeChild(el);
  }, [el, mount]);

  return createPortal(children, el);
};

export default Portal;
