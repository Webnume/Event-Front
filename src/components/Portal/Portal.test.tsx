import { render, fireEvent, screen } from "@testing-library/react";
import ReactDOM from 'react-dom';
import Portal from "./Portal";

describe('Portal', () => {
    it('renders children in a portal', () => {
      const container = document.createElement('div');
      const child = <h1>Portal Content</h1>;
  
      ReactDOM.render(<Portal>{child}</Portal>, container);
  
      expect(container.firstChild).toMatchSnapshot();
    });
  });