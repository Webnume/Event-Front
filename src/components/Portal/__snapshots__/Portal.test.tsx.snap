// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Portal > renders children in a portal 1`] = `null`;
